bigframes/pandas/__init__.py

```diff
@@ -134,6 +138,136 @@ def cut(
+    dummy_na: bool = False,
+    columns: Optional[List] = None,
+    drop_first: bool = False,
+    dtype: Any = None,
+) -> DataFrame:
+    block = data._block
+
+    if isinstance(data, Series):
+        columns = [block.value_columns[0]]
+    if columns is not None and not pandas.api.types.is_list_like(columns):
+        raise TypeError("Input must be a list-like for parameter `columns`")
+    if dtype is not None and dtype not in [
+        pandas.BooleanDtype,
+        bool,
+        "Boolean",
+        "boolean",
+        "bool",
+    ]:
+        raise NotImplementedError(
+            f"Only Boolean dtype is currently supported. {constants.FEEDBACK_LINK}"
+        )
+
+    if columns is None:
+        default_dummy_types = [pandas.StringDtype]
+        columns = [
+            col_id
+            for col_id in block.value_columns
+            if block.expr.get_column_type(col_id) in default_dummy_types
+        ]
+
+    prefix_given = prefix is not None
+
+    def parse_prefix_kwarg(kwarg, kwarg_name):
+        if pandas.api.types.is_list_like(kwarg) and len(kwarg) != len(columns):
+            raise ValueError(
+                f"Length of '{kwarg_name}' ({len(kwarg)}) did not match "
+                f"the length of the columns being encoded ({len(columns)})."
+            )
+        elif pandas.api.types.is_list_like(kwarg):
+            return list(map(str, kwarg))
+        elif isinstance(kwarg, str):
+            return [kwarg] * len(columns)
+        elif isinstance(kwarg, dict):
+            return [kwarg[column] for column in columns]
+        elif kwarg is None:
+            return None
+        else:
+            raise TypeError(f"{kwarg_name} kwarg must be a string, list, or dictionary")
+
+    prefix_sep = parse_prefix_kwarg(prefix_sep, "prefix_sep")
+    if prefix_sep is None:
+        prefix_sep = ["_"] * len(columns)
+    prefix = parse_prefix_kwarg(prefix, "prefix")
+    if prefix is None:
+        prefix = columns
+
+    max_unique_value = (
+        bigframes.core.blocks._BQ_MAX_COLUMNS
+        - len(block.value_columns)
+        - len(block.index_columns)
+        - 1
+    ) // len(columns)
+    columns_ids = []
+    full_prefixes_with_duplicity = []
+    for i in range(len(columns)):
+        label = columns[i]
+        full_prefix = prefix[i] + prefix_sep[i]
+        for col_id in block.label_to_col_id[label]:
+            columns_ids.append(col_id)
+            full_prefixes_with_duplicity.append(full_prefix)
+
+    columns_values = block._get_unique_values(columns_ids, max_unique_value)
+
+    result = block.drop_columns(columns_ids)
+    for i in range(columns_values.nlevels):
+        level = columns_values.get_level_values(i).sort_values()
+        column_label = full_prefixes_with_duplicity[i]
+        column_id = columns_ids[i]
+
+        already_seen = set()
+        for value in level:
+            if pandas.isna(value):
+                continue
+            if value in already_seen:
```

Comment:
If we have generated the unique values properly, we should hit this

Reply:
Removed